olx.py

```diff
@@ -0,0 +1,342 @@
+    out = ""
+    city = city.lower()
+    for char in city:
+        if char == " ":
+            out += "-"
+        elif char in POLISH_CHARACTERS.keys():
+            out += POLISH_CHARACTERS[char]
+        else:
+            out += char
+    return out
+
+
+def url_price_from(price):
+    return "search%5Bfilter_float_price%3Afrom%5D=" + str(price)
+
+
+def url_price_to(price):
+    return "search%5Bfilter_float_price%3Ato%5D=" + str(price)
+
+
+def url_rooms(number):
+    numbers = {1: "one", 2: "two", 3: "three", 4: "four"}
+    return "search%5Bfilter_enum_rooms%5D%5B0%5D=" + numbers[number]
+
+
+def url_yardage_from(minimum):
+    return "search%5Bfilter_float_m%3Afrom%5D=" + str(minimum)
+
+
+def url_yardage_to(maximum):
+    return "search%5Bfilter_float_m%3Ato%5D=" + str(maximum)
+
+
+def url_furniture(furniture):
+    if furniture:
+        return "search%5Bfilter_enum_furniture%5D%5B0%5D=yes"
+    else:
+        return "search%5Bfilter_enum_furniture%5D%5B0%5D=no"
+
+
+def url_floor(floor):
+    return "search%5Bfilter_enum_floor_select%5D%5B0%5D=floor_" + str(floor)
+
+
+def get_url(page=None, *args):
+    url = BASE_URL
+    for element in args:
+        if element is not None:
+            if "filter" in url and "filter" in element:
+                url += element + "&"
+            elif "filter" in element:
+                if page is not None:
+                    url += "?" + page + "&" + element + "&"
+                else:
+                    url += "?" + element + "&"
+            else:
+                url += element + "/"
+    return url
+
+
+def url_buildtype(type):
+    return "search%5Bfilter_enum_builttype%5D%5B0%5D=" + type
+
+
+# TODO: Caching for long urls
+@caching
+def get_content_for_url(url):
+    response = requests.get(url, allow_redirects=False)
+    try:
+        response.raise_for_status()
+    except requests.HTTPError:
+        print('Request for {} failed.'.format(url))
+        return None
+    return response
+
+
+def parse_offer_url(offer_markup):
+    html_parser = BeautifulSoup(offer_markup, "html.parser")
+    url = html_parser.find(class_="linkWithHash").attrs['href']
+    if not url:
+        # detail url is not present
+        return []
+    if urlparse(url).hostname not in WHITELISTED_DOMAINS:
+        # domain is not supported by this backend
+        return []
+    return url
+
+
+def get_title(offer_markup):
+    html_parser = BeautifulSoup(offer_markup, "html.parser")
+    title = html_parser.h1.text
+    return title.replace("\n", "").replace("  ", "")
+
+
+def get_price(offer_markup):
+    html_parser = BeautifulSoup(offer_markup, "html.parser")
+    price = html_parser.find(class_="xxxx-large").text
+    output = ""
+    for char in price:
+        if char.isdigit():
+            output += char
+    return int(output)
+
+
+def get_yardage(offer_markup):
+    html_parser = BeautifulSoup(offer_markup, "html.parser")
+    try:
+        yardage = html_parser.sup.parent.text
```

Comment:
It will raise exception that there is no attribute parent for NoneType (sup) if there is no m^2 symbol in offer